tls/s2n_connection.c

```diff
@@ -283,6 +283,11 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
         return 0;
     }
 
+    if (conn->security_policy_override != NULL) {
```

Comment:
I think this check shouldn't be needed here? It should be checked, but it might make sense to have it closer to where it's dereferenced/used to determine if validation is needed.

Reply:
Are you suggesting that the security policy check should go on the last line right before `conn->config = config`? 
1. The first reason that I didn't want to do that is because `s2n_connection_set_config` mutates a whole bunch of the properties, so if we move the check later and the check fails, it means that the connection will be in an unknown state.
```
connectionA = ...
connectionB = ...

config_bad = ...
config_good = ...

s2n_connection_set_config(connectionA, config_bad) # fails
s2n_connection_set_config(connectionA, config_good) # succeeds

s2n_connection_set_config(connectionB, config_good) # succeeds

# as currently written
connectionA == connectionB

# if we move the check later into the function
connectionA != connectionB
```
That might not be a valuable property, but it seemed like a nice one to have?

2. I think you could make an argument that the first step of applying a config to a connection is determining if the config is valid for a connection. In which case you would check validity as the first step before applying any of the settings.